rust/src/ffi/any.rs

```diff
@@ -215,11 +216,55 @@ impl Downcast for AnyObject {
     }
 }
 
+#[allow(dead_code)]
+pub enum Either<L, R> {
+    Left(L),
+    Right(R),
+}
+
+impl<Q: 'static, A: 'static> Measurement<AnyDomain, Queryable<Q, A>, AnyMetric, AnyMeasure> {
+    pub fn into_any_Q(
+        self,
+    ) -> Measurement<AnyDomain, Queryable<AnyObject, A>, AnyMetric, AnyMeasure> {
+        let function = self.function;
+        Measurement::new(
+            self.input_domain,
+            Function::new_fallible(
+                move |arg: &AnyObject| -> Fallible<Queryable<AnyObject, A>> {
+                    let mut inner_qbl = function.eval(arg)?;
+
+                    Queryable::new(
+                        move |_self, query: Query<AnyObject>| match query {
+                            Query::External(query) => inner_qbl
+                                .eval(query.downcast_ref::<Q>()?)
+                                .map(Answer::External),
+                            Query::Internal(query) => {
+                                if query.downcast_ref::<QueryType>().is_some() {
+                                    return Ok(Answer::internal(Type::of::<Q>()));
+                                }
+                                let Answer::Internal(a) = inner_qbl.eval_query(Query::Internal(query))? else {
+                                    return fallible!(FailedFunction, "internal query returned external answer")
+                                };
+                                Ok(Answer::Internal(a))
+                            }
+                        },
+                    )
```

Comment:
It's interesting that this is basically another form of wrapping (but not with PolyQueryables). I wonder if other instances of this will arise in the future, in which case we might want to have an abstraction if it.